lib/fs.js

```diff
@@ -1621,8 +1625,15 @@ function ReadStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    warnedReadStreamOptions = internalUtil.printDeprecationMessage(
+      'Passing anything but an object or a string as the second argument to ' +
+      'ReadStream is deprecated.',
+      warnedReadStreamOptions
+    );
+    options = options || {};
```

Comment:
My preference is to accept functions as an options property bag option for the time being, so we can keep the "illegal argument!" warning. We can then evaluate whether or not how folks expect a callback to work is consistent and worth supporting as an option.